support/metrics/metricsTracker.go

```diff
@@ -190,34 +188,65 @@ func MakeMetricsTracker(
 	}, nil
 }
 
+// MakeMetricsTrackerGui is a factory method to create a `metrics.Tracker` from the CLI.
+func MakeMetricsTrackerGui(
+	userID string,
+	deviceID string,
+	apiKey string,
+	client *http.Client,
+	botStartTime time.Time,
+	version string,
+	goos string,
+	goarch string,
+	goarm string,
+	guiVersion string,
+) (*MetricsTracker, error) {
+	props := commonProps{
+		CliVersion: version,
+		Goos:       goos,
+		Goarch:     goarch,
+		Goarm:      goarm,
+		GuiVersion: guiVersion,
+	}
+
+	return &MetricsTracker{
+		client:       client,
+		apiKey:       apiKey,
+		userID:       userID,
+		deviceID:     deviceID,
+		props:        props,
+		botStartTime: botStartTime,
```

Comment:
explicitly set all fields on the struct -- i.e. isDisabled and `updateEventSentTime`
should be taken in as an argument like we do for the cli